packages/core/src/forms/form-group/form-group.element.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { LitElement, html } from 'lit-element';
+import { baseStyles, querySlotAll, syncDefinedProps, property, childrenUpdateComplete } from '@clr/core/internal';
+import { CdsInternalControlGroup } from '../control-group/control-group.element.js';
+import { CdsInternalControlInline } from '../control-inline/control-inline.element.js';
+import { CdsControl } from '../control/control.element.js';
+import { FormLayout, ControlWidth } from '../utils/interfaces.js';
+import { getLargestPrimaryLabelWidth } from '../utils/index.js';
+import { styles } from './form-group.element.css.js';
+
+/**
+ * Form
+ *
+ * ```typescript
+ * import '@clr/core/forms/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-form-group layout="horizontal">
+ *   <cds-input>
+ *     <label>Test</label>
+ *     <input type="text" />
+ *   </cds-input>
+ *
+ *   <cds-input>
+ *     <label>Test</label>
+ *     <input type="text" />
+ *   </cds-input>
+ * </cds-form-group>
+ * ```
+ *
+ * @element cds-form
+ * @slot default - For projecting input controls
+ */
+export class CdsFormGroup extends LitElement {
+  /**
+   * @type {horizontal | horizontal-inline | vertical | vertical-inline | compact}
+   * Set to adjust the default control layout for all controls within form group.
+   * When `responsive` is true this will be the largest size to scale to.
+   */
+  @property({ type: String }) layout: FormLayout = 'vertical';
+
+  /**
+   * @type {full | compact}
+   * Adjust the control from the default full width or the browser default width
+   */
+  @property({ type: String }) controlWidth: ControlWidth = 'full';
+
+  /**
+   * By default forms will collapse to layout that prevents overflow.
+   * If disabled control layout may break or overflow in unexpected ways.
+   */
+  @property({ type: Boolean }) responsive = true;
+
+  /**
+   * Set the validate attribute to sync with HTML5 native validation
+   * https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation
+   */
+  @property({ type: Boolean }) validate = false;
+
+  @querySlotAll('[cds-control]') private controls: NodeListOf<CdsControl | CdsInternalControlInline>;
+
+  @querySlotAll('[cds-control-group]') private groups: NodeListOf<CdsInternalControlGroup>;
+
+  private get controlsAndGroups() {
+    return [...Array.from(this.groups), ...Array.from(this.controls)];
+  }
+
+  static get styles() {
+    return [baseStyles, styles];
+  }
+
+  render() {
+    return html`
+      <div class="private-host">
+        <slot cds-layout="vertical gap:lg"></slot>
```

Comment:
When `compact`, we should make this `gap:md` instead of `gap:lg`. Right now, visually, there is no difference in a compact form-group vs. normal.